x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/create_trusted_app_form.tsx

```diff
@@ -96,12 +168,26 @@ export const CreateTrustedAppForm = memo<CreateTrustedAppFormProps>(
         };
       });
     }, []);
+
     const [formValues, setFormValues] = useState<NewTrustedApp>({
       name: '',
       os: 'windows',
       entries: [generateNewEntry()],
       description: '',
     });
+
+    const [validationResult, setValidationResult] = useState<ValidationResult>(() =>
```

Comment:
when are we using react's state vs redux state?

Reply:
😝 answer here 👉  https://www.elastic.co/about/our-source-code#it-depends

___
😄 to expand on my reasoning:

For this one - the entire thing - it was developed as a "pure" component and it is not tied to the Store. I did not see the benefit of storing this in Redux while the data is being collected from the user.  Another reason, but one that really would probably not have convinced me to change the approach, was that at the time I started it, the trusted apps store setup was not yet available (pending PR).